src/firewall/iptables.rs

```diff
@@ -37,18 +37,34 @@ pub fn new() -> Result<Box<dyn firewall::FirewallDriver>, Box<dyn Error>> {
 
 impl firewall::FirewallDriver for IptablesDriver {
     fn setup_network(&self, network_setup: SetupNetwork) -> Result<(), Box<dyn Error>> {
+        let mut interface = "".to_string();
+        if let Some(_int) = network_setup.net.network_interface.clone() {
+            match network_setup.net.network_interface {
+                Some(iface) => interface = iface,
+                None => {
+                    return Err(Box::new(crate::error::NetavarkError {
+                        error: "failed to get interface".to_string(),
+                        errno: 1,
+                    }));
+                }
+            }
+        }
```

Comment:
This looks very complex, i think you just need the inner match block with `let interface = match ...`